Source/API/Extras/CBLIncrementalStore.m

```diff
@@ -2057,103 +2090,122 @@ - (void) processCouchbaseLiteChanges: (NSArray*)changes {
         NSManagedObjectID* objectID = [self newObjectIDForEntity: entity referenceObject: reference];
 
         if (deleted) {
-            [deletedObjectIDs addObject: objectID];
-        } else {
-            [updatedObjectIDs addObject: objectID];
+            [deletedIDs addObject: objectID];
+        }
+        else {
+            [updatedIDs addObject: objectID];
         }
+
     }
 
-    for (NSManagedObjectContext* context in self.observingManagedObjectContexts) {
-        [self informManagedObjectContext: context
-                              updatedIDs: updatedObjectIDs deletedIDs: deletedObjectIDs];
+    if (updatedIDs.count > 0 || deletedIDs.count > 0) {
+        [self refreshContextsWithUpdatedIDs:updatedIDs deletedIDs: deletedIDs];
+
+        [[NSNotificationCenter defaultCenter]
+         postNotificationName: kCBLISObjectHasBeenChangedInStoreNotification
+         object: self
+         userInfo: @{ NSUpdatedObjectsKey: updatedIDs,
+                      NSDeletedObjectsKey: deletedIDs }];
```

Comment:
Notification should be fired after all refresh operations are finished. Note that refreshContextsWithUpdatedIDs: deletedIDs: is actually done in the background.